options.js

```diff
@@ -29,6 +29,11 @@ module.exports.restoreDefaultOptions = () => {
 module.exports.getMergedOptions = options => {
   const opts = merge({}, defaultOptions, options);
 
+  // If user provided target element we mark it so we can then query its offset in puppeteer
+  if (options && options.target) {
+    options.target.setAttribute("data-jsdom-screenshot-target", "");
```

Comment:
Instead of setting the attribute here and instead of accepting a `target` as a DOM node, we could: ccept a string as `target` which we then pass to `document.querySelector` in `generateImage.js` as `const target = document.querySelector(opts.target);`.

This means we wouldn't need to mangle the DOM.

What do you think?

Reply:
I thought about that, but in that case, we will be forcing the user to set a (unique) identification in its own code to match that element. In general I wouldn't not like this and it will not be the case. For example, I'm working with styled-components / emotion to implement the styles and these tools are auto generating class names for me so I rarely set ids or classes in my markup and the generated class names are unknown. 

Probably your suggestion will make our side easier/safer but won't be so comfortable for users that, when working with jsdom they will be working directly with DOM elements directly and probably they won't have any identificator. 

If you disagree I could change the implementation, but I would like the target API to be like it's now. Alternatively we could accept both an element or a selector but not only a selector. Hope you get my point.

Reply:
Thanks for explaining. What I dislike about accepting an element is that a function which is supposed to take a screenshot mutates the JSDOM. This is unexpected and can break other tests (e.g. code-snapshot tests). This is a no-go for me.

I have trouble following the argument about ids/classes. Devs need to find their target element one way or another in any case. The downside of using a string to pass to `document.querySelector` instead of accepting an element is that it's no longer possible to use regular JS to find the element, the element needs to be describable by CSS selector string. That's not too bad though.

If people really need an escape hatch, they can always set the attribute manually as you're doing right now with `element.setAttribute("data-jsdom-screenshot-target", "")` and then use `[data-jsdom-screenshot-target]` as the CSS selector string. This makes accepting a CSS selector string cover all cases of accepting an element and therefore more powerful (and the mutation becomes explicit in test-code).

For these reasons I'd like to propose accepting CSS selector string only. Hope that made sense

Reply:
OK, make sense, I tried to use a "unique" data attribute but you are right, is not OK the function modify the DOM. What if after the screenshot I remove the data-attribute, would that be acceptable ? Also, I wonder if it's possible we could reference the element with xpath without modifying the DOM? Nevertheless, I will try to make the changes you suggest. Thanks